tasks/volumes.yml

```diff
@@ -27,11 +27,26 @@
     {% endif %}
     -a {{ ansible_check_mode }}
   with_items: "{{ volumes }}"
-  when: item.type | default(libvirt_volume_default_type) == 'volume'
+  when: "{{ item.device | default(libvirt_volume_default_device) == 'disk' }}"
   environment: "{{ libvirt_vm_script_env }}"
   register: volume_result
   changed_when:
     - volume_result is success
     - (volume_result.stdout | from_json).changed | default(True)
   check_mode: False
   become: true
+
+# TODO: If the image already exists will this command reformat it while in use?
```

Comment:
Does this need to be done before the libvirt volume is created?

Reply:
I believe this behaves but still need to explicitly double check.

Reply:
To answer your question: the volume can be created any time before the instance is.

To answer the TODO note: volumes are not reformatted on subsequent runs.